libtransmission/rpcimpl.cc

```diff
@@ -442,6 +442,7 @@ namespace make_torrent_field_helpers
         peer_map.try_emplace(TR_KEY_clientIsChoked, peer.clientIsChoked);
         peer_map.try_emplace(TR_KEY_clientIsInterested, peer.clientIsInterested);
         peer_map.try_emplace(TR_KEY_clientName, peer.client);
+        peer_map.try_emplace(TR_KEY_peer_id, peer.peer_id);
```

Comment:
Precaution: Please base64-encode the peer ID in case there's non-UTF8 byte sequences (JSON can ***only*** contain UTF8 byte sequences by definition).

Reply:
Keep the value in `tr_peer_stat` raw, because it's not just used for RPC. Encode in the line I marked for this review comment instead.

Reply:
Just for a bit of context, the C types are there just for historical reasons, nothing to worry about. Ultimately they should all be phased out in our code bsae.